transport/http/server/server.go

```diff
@@ -139,7 +144,24 @@ func NewServer(cfg config.ServiceConfig, handler http.Handler) *http.Server {
 	return NewServerWithLogger(cfg, handler, nil)
 }
 
+type serverOptions struct {
+	// UseH2C enable h2c support.
+	UseH2C bool `json:"use_h2c"`
+}
+
 func NewServerWithLogger(cfg config.ServiceConfig, handler http.Handler, logger logging.Logger) *http.Server {
+	// wrap handler with h2c handler if it is enabled
+	srvCfg := serverOptions{}
+	if v, ok := cfg.ExtraConfig[ginNamespace]; ok {
+		if d, err := json.Marshal(v); err == nil {
+			if err := json.Unmarshal(d, &srvCfg); err == nil {
+				if srvCfg.UseH2C {
```

Comment:
We can remove this, by adding a `UseH2C` field in the [ServiceConfig](https://github.com/luraproject/lura/blob/master/config/config.go#L44), (perhaps close to the [TLS](https://github.com/luraproject/lura/blob/master/config/config.go#L158) in the code, as it is kind of related with HTTP2). 
That will remove the need of having the `serverOptions` in this file, and the `ginNamespace`.

For the "backport" of the "github_com/luraproject/lura/router/gin", we can add code in krakend-ce once merged that emits a deprecation warning of that option, and "moves" the value to this new top level flag.